### DISCUSSION TITLE
请问如何修改icarus主题分类排序

### DISCUSSION THREAD

**Discussion by nanbowanya** [ppoffice/hexo-theme-icarus#1177] [Q&A]
开发者你好，hexo使用的icarus主题。希望自定义分类顺序。请问该如何修改？

**Comment by ppoffice** [comment]
@nanbowan323 https://github.com/hexojs/hexo-generator-category#options

**Reply by nanbowanya** [reply]
😂😂😂我用的icarus4.0，尝试过上述方式在_config.icarus.yml和_config.yml中添加均不会对分类排序产生变化🙁

**Reply by ppoffice** [reply]
你添加的什么配置？

**Reply by nanbowanya** [reply]
> 你添加的什么配置？

category_generator:
  per_page: 10
  order_by: -date

还有
category_generator:
  per_page: 5
  order_by: -length
都尝试过，而且上述方式好像只能通过时间，数量修改。我更希望自定义，这篇文章有介绍，但他是ejs的，https://www.hongfs.cn/2020/06/javascript/hexo-theme-icarus-category-sort/

**Comment by ppoffice** [comment]
那需要对这里的结果进行排序：https://github.com/ppoffice/hexo-component-inferno/blob/d965a998b4667c6cdbd11436a3e23843b4d12b26/src/view/widget/categories.jsx#L120-L123

至于如何修改这个文件，参考：https://ppoffice.github.io/hexo-theme-icarus/uncategorized/%E5%B8%B8%E8%A7%81%E9%97%AE%E9%A2%98/#%E5%B8%83%E5%B1%80
